public/static/js/spawned_hosts.js

```diff
@@ -166,12 +167,26 @@ mciModule.controller('SpawnedHostsCtrl', ['$scope', '$window', '$timeout', '$q',
       $scope.updateSpawnedHosts();
     }, 60000);
 
-    // Returns true if the user can spawn another host. If hosts has not been initialized it
+    // Returns true if the user can spawn another host. If hosts have not been initialized it
     // assumes true.
     $scope.availableHosts = function () {
       return ($scope.hosts == null) || ($scope.hosts.length < $scope.maxHostsPerUser)
     }
 
+    $scope.availableUnexpirableHosts = function () {
+      var available = $scope.maxUnexpirableHostsPerUser;
+      for (host of $scope.hosts) {
+        if (host.original_expiration == null) {
+          available--
+        }
+      }
+      return available
+    }
+
+    $scope.noExpirationEnabled = function () {
```

Comment:
Can this function be renamed? It's easy to read as in "there is not expiration enabled" which is kind of confusing

Reply:
done.
renamed to `unexpirableEnabled`